pkg/purl/purl.go

```diff
@@ -198,13 +198,25 @@ func (p *PackageURL) BOMRef() string {
 	return purl.String()
 }
 
+// NewPackageIdentifier creates a new package identifier based on PURL for the given package.
+func NewPackageIdentifier(target ftypes.TargetType, pkg ftypes.Package) ftypes.PkgIdentifier {
+	pkgURL, err := NewPackageURL(target, types.Metadata{}, pkg)
+	if err != nil || pkgURL == nil || pkgURL.Type == "" {
```

Comment:
If I remember correctly, I removed all cases where we could get `pkgURL != nil && pkgURL.Type == ""`.
we can remove `pkgURL.Type == ""` check.

Reply:
done at https://github.com/aquasecurity/trivy/pull/5439/commits/5e72c5a95058110fdaeac484791abac7d35a288e